index.js

```diff
@@ -44,49 +34,126 @@ app.get('/docs', (req, res) => {
   res.send(apiDocs);
 });
 
-app.get('/status', (req, res) => {
-  res.send({
-    'statusCode': wallet.state,
-    'statusMessage': humanState[wallet.state],
-    'network': wallet.network,
-    'serverUrl': wallet.server,
-    'serverInfo': wallet.serverInfo,
+app.post('/start', (req, res) => {
+  // We expect the user to send the seed he wants to use
+  // if not sent, we use the 'default' key
+  const seedKey = req.body.seedKey || 'default';
+  if (!(seedKey in config.seeds)) {
+    res.send({
+      success: false,
+      message: 'Seed not found.',
+    });
+    return;
+  }
+
+  // The user must send a key to index this wallet
+  if (!('key' in req.body)) {
+    res.send({
+      success: false,
+      message: 'Parameter \'key\' is required.',
+    });
+    return;
```

Comment:
If we go this way of having the key in the body, I *think* there's a way to use an express middleware to handle it, so all paths that need the key go through that first.

But as we want this done by tomorrow, I'm ok if we merge like this and refactor later.

Reply:
Even if we go with other implementations it's important to use this middleware to handle the required parameters.

I am already doing it here (https://github.com/HathorNetwork/hathor-wallet-headless/blob/e830b4998b121e79a06373fefbe857ca1125b7dc/index.js#L94) in the `walletRouter`, so all `/wallet/...` requests don't need to check it. However, this `/start/` endpoint you added the comment is not inside `walletRouter` and I can't have it handled by a global middleware because I have other endpoints (`/` and `/docs`) that don't need the parameter.

If you know anyway I could do it better, I can refactor it.

Reply:
You're right, it was my bad. I confused this other piece of code to another endpoint, so I thought we were repeating it. 

The question is why this api is not under `walletRouter`? Why don't we use `/wallet/start`?

Reply:
I defined the API's under `walletRouter` to be the ones where you already have a wallet created and want to execute an action on it. It was just the way I thought at first, the `/start/` would be to create but I can move to `/wallet/start/`

Reply:
I think it makes sense semantically (to have it under `/wallet/`). So we have

```
/wallet/start
/wallet/stop
/wallet/status
```

Reply:
Actually to me semantically makes more sense to have under `/wallet/` some actions to be done in a wallet that you already have. And the code becomes less complex, since I can create the middleware that I already have https://github.com/HathorNetwork/hathor-wallet-headless/blob/e830b4998b121e79a06373fefbe857ca1125b7dc/index.js#L94

The piece of code below

```
  if (!(key in wallets)) {
    sendError('Invalid key parameter.')
    return;
  }

  const wallet = wallets[key];
  if (!wallet.isReady()) {
    sendError('Wallet is not ready.', wallet.state)
    return;
  }

  // Adding to req parameter, so we don't need to get it in all requests
  req.wallet = wallet;
```

is executed to all requests under `/wallet/`. I would need to add an extra check to execute this only if it's not a `/start/` because the context of `/wallet/start` is totally different from the other requests under `/wallet/`. That's why I left `/start/` outside `/wallet/`.

Reply:
Got it. I'm ok with it them.